### PR TITLE
fix: dev: #222: Add validation to check for corrupted cached file

### PR DESCRIPTION
**Pull request by abhishekdas99**

### CODE REVIEW

File: rubix-bookkeeper/src/test/java/com/qubole/rubix/bookkeeper/validation/TestFileValidatorVisitor.java [hunk 1]
```diff
@@ -53,7 +77,10 @@ public void tearDown() throws IOException
     CacheConfig.setCacheDataDirPrefix(conf, TEST_CACHE_DIR_PREFIX);
     TestUtil.removeCacheParentDirectories(conf, TEST_MAX_DISKS);
 
+    BaseServerTest.stopBookKeeperServer();
+
     conf.clear();
+    metric = null;
```
Comment (jordanw-bq) [inline rubix-bookkeeper/src/test/java/com/qubole/rubix/bookkeeper/validation/TestFileValidatorVisitor.java]: `metric` is reassigned at the start of every test, so this shouldn't be necessary. However, this does make it clearer that the metrics registry is reset every time.

File: rubix-bookkeeper/src/test/java/com/qubole/rubix/bookkeeper/validation/TestFileValidatorVisitor.java [hunk 2]
```diff
@@ -122,6 +149,106 @@ public void testValidatorFileVisitor_someMdExists_multiDepth() throws IOExceptio
     runAndVerifyValidator(conf, maxDepth, maxDirs, maxFiles, mdSkip);
   }
 
+  @Test
+  public void testValidatorFileVisitor_allValidCachedFiles() throws TException, IOException
```
Comment (jordanw-bq) [inline rubix-bookkeeper/src/test/java/com/qubole/rubix/bookkeeper/validation/TestFileValidatorVisitor.java]: There's a lot of repeated code in these tests for creating the file + getting the status + setting the blocks as cached. Would it be possible to refactor these steps into another method?

Comment (abhishekdas99) [reply rubix-bookkeeper/src/test/java/com/qubole/rubix/bookkeeper/validation/TestFileValidatorVisitor.java]: Done

Comment (jordanw-bq) [reply rubix-bookkeeper/src/test/java/com/qubole/rubix/bookkeeper/validation/TestFileValidatorVisitor.java]: LGTM

**Comment by jordanw-bq** [comment]
@abhishekdas99 Latest commit (7dc58a0) LGTM

**Comment by jordanw-bq** [comment]
Same for the test refactoring

**Pull request closed without merging**